### PR TITLE
feat(localization): Add private method to access bundles

### PR DESCRIPTION
**Pull request by bretkikehara**
Add private method to access bundles.

**use case:** Load bundles for localization management

[Review on Reviewable](https://reviewable.io/reviews/doshprompt/angular-localization/82)

### CODE REVIEW

File: src/localization.js [hunk 1]
```diff
@@ -287,6 +310,11 @@ angular.module('ngLocalize')
             getKey: getKey,
             setLocale: setLocale,
             getLocale: getLocale,
-            getString: getLocalizedString
+            getString: getLocalizedString,
+            $$bundleReady: function (path) {
```
Comment (doshprompt) [inline src/localization.js]: don't need this anymore imo as you can listen in on `localeEvents.resourceUpdates`

File: src/localization.js [hunk 2]
```diff
@@ -266,7 +294,6 @@ angular.module('ngLocalize')
                 updateHtmlTagLangAttr(lang);
 
                 $rootScope.$broadcast(localeEvents.localeChanges, currentLocale);
-                $rootScope.$broadcast(localeEvents.resourceUpdates);
```
Comment (bretkikehara) [inline src/localization.js]: If resourceUpdates event will pass back the bundle objects, then this needs to be removed as bundles are not loaded here.

File: tests/unit/serviceSpec.js [hunk 3]
```diff
@@ -74,5 +74,21 @@ describe('service', function () {
             });
             $httpBackend.flush();
         }));
+
+        it('should freeze bundle', inject(function (locale, $httpBackend) {
+            var resp = 'lang file not found';
+            $httpBackend.expectGET('languages/en-US/common.lang.json').respond(200, {
+                "yes": "Yes"
+            });
+            locale.$$bundleReady('common').then(function (bundle) {
```
Comment (doshprompt) [inline tests/unit/serviceSpec.js]: Does this still run? The test should fail as $$bundleReady no longer available. Also, need to change it to check the event.

File: src/localization.js [hunk 4]
```diff
@@ -92,9 +114,15 @@ angular.module('ngLocalize')
 
                             // Mark the bundle as having been "loaded".
                             delete root._loading;
+                            parent[ref] = freeze(root);
+                            root = null;
 
                             // Notify anyone who cares to know about this event.
-                            $rootScope.$broadcast(localeEvents.resourceUpdates);
+                            $rootScope.$broadcast(localeEvents.resourceUpdates, {
```
Comment (doshprompt) [inline src/localization.js]: NICE!

File: tests/unit/serviceSpec.js (hunk 3 above)
Comment (bretkikehara) [inline tests/unit/serviceSpec.js]: The service tests were updated to reflect the frozen bundle object

**Comment by doshprompt** [comment]
Looks good, but just a thought, do we need locale in there? Can't we do `locale.getLocale()`

Also, what if we broadcast `path: bundle`

i.e. 

```json
"common": {
    "yes": "YES",
    "no": "NO"
}
```

**Comment by bretkikehara** [comment]
No, `locale.getLocale()` is a synchronous while the `resourceUpdates` event is async due to the $http request. Although the `locale.getLocale()` will suffice for most cases, the call will be incorrect when loading multiple locales in quick succession.

In addition, `path: bundle` would not work because the event has no knowledge of the path that returned its $http response.

**Consider:**
```javascript
angular.forEach([ 'en-US', 'fr-FR' ], function (lang) {
    locale.setLocale(lang);
    locale.ready([ 'common', 'uncommon', 'rare' ]);
});

$scope.$on(localeEvents.resourceUpdates, function (e, data) {
    if (!$scope.localeCache) {
        $scope.localeCache = {};
    }
    if (!$scope.localeCache[data.locale]) {
        $scope.localeCache[data.locale] = {};
    }
    $scope.localeCache[data.locale][data.path] = angular.copy(data.bundle);
});
```

**Pull request merged**